Entities/Industry/Saw/Saw.as

```diff
@@ -152,8 +152,11 @@ void Blend(CBlob@ this, CBlob@ tobeblended)
 
 bool canSaw(CBlob@ this, CBlob@ blob)
 {
-	if (blob.getRadius() >= this.getRadius() * 0.99f || blob.getShape().isStatic() ||
-	        blob.hasTag("sawed") || blob.hasTag("invincible"))
+	if (blob.getRadius() >= this.getRadius() * 0.99f 
+		|| blob.getShape().isStatic() 
+		|| blob.hasTag("sawed") 
+		|| blob.hasTag("invincible")
+		|| (!blob.hasTag("can be sawed immediately") && blob.getTickSinceCreated() < 30)) // fix for getting insta-killed after changing class
```

Comment:
30 ticks seems too generous as it would be possible to (accidentally) abuse this by coming into contact with a saw shortly after switching classes (could reasonably happen even after TPing into a saw). I'm not sure if 5 fully solves the problem the PR aims to resolve (e.g. due to lag), but let's try. FWIW, the fact that you can get sawed from class swap is more of an engine bug than anything.

```suggestion
		|| (!blob.hasTag("can be sawed immediately") && blob.getTickSinceCreated() < 5)) // fix for getting insta-killed after changing class
```